pkg/apis/cilium/v1alpha1/types_network.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type IdentityAllocationMode string
+
+const (
+	CRD     IdentityAllocationMode = "crd"
+	KVStore IdentityAllocationMode = "kvstore"
+)
+
+type TunnelMode string
+
+const (
+	VXLan    TunnelMode = "vxlan"
+	Geneve   TunnelMode = "geneve"
+	Disabled TunnelMode = "disabled"
+)
+
+type KubeProxyReplacementMode string
+
+const (
+	Strict  KubeProxyReplacementMode = "strict"
+	Probe   KubeProxyReplacementMode = "probe"
+	Partial KubeProxyReplacementMode = "partial"
+)
+
+type NodePortMode string
+
+const (
+	Hybird NodePortMode = "hybrid"
+)
+
+type Store string
+
+const (
+	Kubernetes Store = "kubernetes"
+	ETCD       Store = "etcd"
+)
+
+// Prometheus configuration for cilium
+type Prometheus struct {
+	Enabled bool  `json:"prometheusEnabled"`
+	Port    int32 `json:"port"`
+}
+
+// OperatorPrometheus configuration for cilium
+type OperatorPrometheus struct {
+	Enabled bool  `json:"operatorprometheusEnabled"`
+	Port    int32 `json:"port"`
+}
+
+// ExternalIPs configuration for cilium
+type ExternalIP struct {
+	// ExternalIPenabled is used to define whether ExternalIP address is required or not.
+	Enabled bool `json:"externalipEnabled"`
+}
+
+// Hubble enablement for cilium
+type Hubble struct {
+	Enabled bool     `json:"enabled"`
+	UI      bool     `json:"ui"`
+	Metrics []string `json:"metrics"`
+}
+
+// Nodeport enablement for cilium
+type Nodeport struct {
+	// Enabled is used to define whether Nodeport is required or not.
+	Enabled bool `json:"nodePortEnabled"`
+	// Mode is the mode of NodePort feature
+	Mode NodePortMode `json:"nodePortMode"`
```

Comment:
It doesn't make sense to have both `Mode` and `Enabled`. Maybe change it so there is a `NodePortMode=off`

Reply:
This is to match the charts and the documentation from Cilium. So from the usability perspective I would suggest we keep it as is.